deep_qa/common/params.py

```diff
@@ -12,6 +31,28 @@ def _get_choice_error_message(value: Any, choices: List[Any], name: str=None) ->
         return '%s not in acceptable choices: %s' % (value, str(choices))
 
 
+def pop_with_logging(params: Dict[str, Any], key: str):
```

Comment:
We actually need more detail than this.  We need a `name`, here, which tells you which class, or (even better) which path gave us this parameter.  Ideally, you could handle this like namespaces are handled in tensorflow, just adding a new namespace when you start reading parameters for an object you're creating, so that things are automatically nested and you can basically read off the entire nested json structure.  Does this make sense?